src/sgame/sg_bot_ai.cpp

```diff
@@ -788,6 +788,29 @@ AINodeStatus_t BotActionSay( gentity_t *self, AIGenericNode_t *node )
 	return STATUS_SUCCESS;
 }
 
+// Check if a human bot should switch to blaster to attack an offmesh target.
+// TODO: Add different ranges for buildings vs moving targets.
+//       If the target can move, we could allow a higher range for spready guns.
+// This function has a small overlap with BotTargetInAttackRange.
+static bool TargetInOffmeshAttackRange( gentity_t *self )
```

Comment:
Maybe you can extract like a GetEffectiveWeaponRange function out of BotTargetInAttackRange?

Reply:
if we really wanted to make things clean on the weapon range codes (this is not the only place), they we'd need to move those values from hard-coded into the weaponAttributes_t, so that we'd be able to get rid of that kind of mess everywhere (same goes for obituaries actually).
I remember doing efforts in that direction a while ago, but it was not well received...

Reply:
There is not really code duplication, but rather a conceptual overlap. I do not think this should be blocking, as it is commented and not hidden in any way. I agree that we should have a closer look at BotTargetInAttackRange, but let's do this in a seperate PR.

Reply:
That's OK but please add a comment about why the ranges might need to be different.